web/src/user_group_edit.ts

```diff
@@ -1541,28 +1564,84 @@ function realm_has_deactivated_user_groups(): boolean {
     return deactivated_group_count > 0;
 }
 
-export function update_empty_left_panel_message(): void {
+export function update_empty_left_panel_message(is_searching?: boolean): void {
     // Check if we have any groups in panel to decide whether to
     // display a notice.
-    let has_groups;
     const is_your_groups_tab_active =
         get_active_data().$tabs.first().attr("data-tab-key") === "your-groups";
+    let current_group_filter =
+        filters_dropdown_widget.value() ?? FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
+    if (is_searching) {
+        let selector = "";
+        if (current_group_filter === FILTERS.DEACTIVATED_GROUPS) {
+            selector = ".deactivated-group";
+        } else if (current_group_filter === FILTERS.ACTIVE_GROUPS) {
+            selector = ":not(.deactivated-group)";
+        }
+        const groups = $(".user-groups-list").children(selector).length;
```

Comment:
Can this selector directly use `hidden` attrbute to check if there are any visible user grups in the list or not?

Reply:
And I think that can just be used generally as well for other filters as well and will make the code much more simpler.